### PR TITLE
[feature] Add dispatcher and pin handler, reorganize codebase and logging

### PR DESCRIPTION
**Pull request by brendacs**
### Purpose of pull request

**Related Issue**: N/A

Add event emitter to dispatch messages to various sections of the bot.

### Pull request checklist

- [x] Branch and PR are named correctly
- [ ] Updated relevant documentation
- [x] Tested with a tester bot
- [x] Wrote unit tests for changes

### Testing instructions

Run the bot and try a command with a sub command like ++pins list. The logger should log the sub command if it gets passed to the pins' handler.

### CODE REVIEW

File: src/bots/pins/index.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import client from '../../client';
+import handler from './handler';
+
+client.on('pins', (args) => handler(args));
```
Comment (sirNikolai) [inline src/bots/pins/index.js:4]: Should we create a healthcheck minibot?
Just literally a ping command for this sort of occasion.

File: src/utils/dispatcher.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    const args = msg.substring(2).split(' ');
+    const cmd = args[0];
+    if (validCmds.includes(cmd)) {
+      client.emit('command', cmd, args.splice(1));
+    }
+  }
+};
+
+export const dispatchCmd = (cmd, args) => {
+  // TODO: check perms
```
Comment (sirNikolai) [inline src/utils/dispatcher.js:16]: would this be bot specific though?

File: src/utils/tests/index.test.js [hunk 3]
```diff
@@ -1,16 +1,3 @@
...
-
-  test('returns subcmd when present', () => {
-    expect(parseMessage('++faq LC')).toEqual({ cmd: 'faq', subCmd: 'LC' });
-  });
-
-  test('undefined when cmd invalid', () => {
-    expect(parseMessage('++invalid')).toEqual(undefined);
-  });
+test('tester test', () => {
+  expect(true).toBe(true);
```
Comment (sirNikolai) [inline src/utils/tests/index.test.js:2]: Can we remove the file in this case?

**Review by sirNikolai — APPROVED** [review]
lgtm

left some comments that are some talking points more than strict change requests

File: src/utils/dispatcher.js (hunk 2 above)
Comment (brendacs) [reply src/utils/dispatcher.js:16]: that's true i'll remove this

**Pull request merged**